### PR TITLE
Add missing AWS feature scripts and utilities for running in Jenkins jobs

### PR DESCRIPTION
**Pull request by jdotpz**

### CODE REVIEW

File: bin/jenkins/jenkinslib/identify_infra_to_update.sh [hunk 1]
```diff
@@ -0,0 +1,189 @@
+function identify_infra_to_update() {
+    AUTOSCALENAME="";ELBNAME=""
+    case ${ROLEENVNAME} in
+        webapp-stage )
```
Comment (rhelmer) [inline bin/jenkins/jenkinslib/identify_infra_to_update.sh]: I worry about it being hard to maintain lists of roles and mappings to other names etc. in different places, not enough to block just something that occurs to me when I see repetition :)

Comment (jdotpz) [inline bin/jenkins/jenkinslib/identify_infra_to_update.sh]: Hmmm, that's a very good point.  I suppose I don't see any reason why I can't put all the identify mappings for roles into a single script.  I will need different .list files to identify things to update versus things to kick, but that is better than separate role mapping files.

**Comment by jdotpz** [comment]
Squashed!

File: puppet/README.md [hunk 2]
```diff
@@ -42,3 +42,113 @@ if desired.
...
+
+
+## Setup for Buildbox (Jenkins)
+
+Additional steps are required to prepare Jenkins to properly use the
+aforementioned deployment scripts.
+
+### Jenkins IAM user
+
+<<<<<<< HEAD
```
Comment (phrawzty) [inline puppet/README.md]: Looks like a squash artefact - please remove.

File: puppet/README.md [hunk 3]
```diff
@@ -42,3 +42,113 @@ if desired.
...
+
+RPMs for both Terraform and Packer are available in the `socorro-public` Yum
+repo and can be installed in the usual fashion.
+
+```bash
+sudo yum install packer terraform
+```
+
+*TODO*: Config details.
+* Put the signing key in ```/home/centos/.rpmsign
```
Comment (phrawzty) [inline puppet/README.md]: Please fix this markdown.

**Comment by phrawzty** [comment]
`r+` with nits.

# :tada: :cake: :icecream:

**Comment by rhelmer** [comment]
r+ this all tested ok

File: bin/deploy-socorro.sh [hunk 4]
```diff
@@ -79,29 +103,31 @@ function scale_in_per_elb() {
     aws autoscaling set-desired-capacity --auto-scaling-group-name ${AUTOSCALENAME} --desired-capacity ${DEPLOYCAPACITY}
          RETURNCODE=$?;error_check
     echo "`date` -- Desired capacity set with a return code of ${RETURNCODE}"
+    echo "`date` -- Suspending scaling processes until we're ready to scale away"
+    #sleep 10  # Give AWS time to scale in before we turn off scaling
```
Comment (rhelmer) [inline bin/deploy-socorro.sh]: want to just remove this if it's commented out anyway?

File: bin/deploy-socorro.sh [hunk 5]
```diff
@@ -79,29 +103,31 @@ function scale_in_per_elb() {
...
+    #aws autoscaling suspend-processes --auto-scaling-group-name ${AUTOSCALENAME}
```
Comment (rhelmer) [inline bin/deploy-socorro.sh]: sam

**Comment by rhelmer** [comment]
r+ lgtm

**Comment by jdotpz** [comment]
Pretty?  No.  Functional?  Yes.

**Comment by phrawzty** [comment]
Merge when ready good sirs.

**Comment by phrawzty** [comment]
:100:

**Pull request merged**